extensions/resource-deployment/src/services/tools/azCliTool.ts

```diff
@@ -40,18 +42,98 @@ export class AzCliTool extends ToolBase {
 		return true;
 	}
 
-	public install(): Promise<void> {
-		return Promise.reject('not implemented');
+	protected async getInstallationPath(): Promise<string | undefined> {
+		switch (this.osType) {
+			case OsType.linux:
+				return defaultInstallationRoot;
+			default:
+				return win32InstallationRoot;
+		}
 	}
 
+	readonly allInstallationCommands: Map<OsType, Command[]> = new Map<OsType, Command[]>([
+		[OsType.linux, linuxInstallationCommands],
+		[OsType.win32, win32InstallationCommands],
+		[OsType.darwin, macOsInstallationCommands],
+		[OsType.others, defaultInstallationCommands]
+	]);
+
 	protected getVersionFromOutput(output: string): SemVer | undefined {
 		if (output && output.includes('azure-cli')) {
 			return new SemVer(output.split(EOL)[0].replace('azure-cli', '').replace(/ /g, '').replace('*', ''));
 		} else {
 			return undefined;
 		}
 	}
-	protected get versionCommand(): string {
-		return 'az --version';
+	protected get versionCommand(): Command {
+		return {
+			command: 'az --version'
+		};
 	}
 }
+
+const win32InstallationCommands = [
+	{
+		comment: localize('resourceDeployment.AziCli.DeletingPreviousAzureCli.msi', 'deleting previously downloaded azurecli.msi if one exists ...'),
```

Comment:
I am not sure whether these strings need to be localized, this is for dev debugging purpose, better keep it English for our own benefit :)

Reply:
It is also for the benefit of the user to debug and correct what went wrong. We will know from context what it is even if it is in some other language, so being friendlier to the end user is more useful. Remember when installation commands fail for unforeseen reasons, expectation is that user will look at output channel, figure out what went wrong, corrrect offline and the retry installation